api2-war/src/main/java/eu/europeana/api2/v2/utils/ApiKeyUtils.java

```diff
@@ -66,21 +77,90 @@ public LimitResponse checkLimit(String wskey, String url, RecordType recordType,
+            apiKey = createTempApiKey(wskey);
+        }
+        return new LimitResponse(apiKey, requestNumber);
+    }
+
+    /**
+     * Validates the supplied apikey using the apikey service
+     * <p>
+     * NOTE: for the time being, this method mimics the user response of the old checkLimit() method with regards to
+     * the JSON format of the output, the returned requestnumber and messages and format of possible error conditions,
+     * including providing a temporary validated key if the apikey service cannot be reached
+     *
+     * @param key The user's public apikey
+     * @return A {@link LimitResponse} consisting of the apiKey and current request number
+     * @throws ApiKeyException {@link ApiKeyException} if an unregistered or unauthorised key is provided
+     *
+     */
+    public LimitResponse validateApiKey(String key) throws ApiKeyException {
+        if (StringUtils.isBlank(key)) {
+            throw new ApiKeyException(ProblemType.APIKEY_MISSING, null, HttpStatus.SC_BAD_REQUEST);
+        }
+
+        ApiKey apiKey;
+        long requestNumber;
+
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpPost            httpPost = new HttpPost(urlService.getApikeyValidateUrl());
+            httpPost.setHeader(AUTHORIZATION, "APIKEY " + key);
+            long                  t        = System.currentTimeMillis();
+            CloseableHttpResponse response = client.execute(httpPost);
+            LOG.debug("Post request to validate apiKey took {} ms", (System.currentTimeMillis() - t));
+
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NO_CONTENT) {
+                LOG.debug("Apikey validated");
+                requestNumber = 999;
+                apiKey = createApiKey(key);
+                apiKey.setApiKey(key);
+            } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_UNAUTHORIZED) {
+                LOG.error("Apikey is not registered");
+                throw new ApiKeyException(ProblemType.APIKEY_NOT_REGISTERED, key, response.getStatusLine().getStatusCode());
+            } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_GONE) {
+                LOG.error("Apikey is deprecated");
+                throw new ApiKeyException(ProblemType.APIKEY_DEPRECATED, key, response.getStatusLine().getStatusCode());
+            } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
+                LOG.error("No Apikey provided");
+                throw new ApiKeyException(ProblemType.APIKEY_MISSING, key, response.getStatusLine().getStatusCode());
+            } else {
+                LOG.error("{}: Apikey service returned HTTP status {}", UNEXPECTED,
+                          response.getStatusLine().getStatusCode());
+                throw new ApiKeyException(ProblemType.APIKEY_OTHER, key, response.getStatusLine().getStatusCode());
+            }
+
+        } catch (IOException e) {
```

Comment:
I think we will need to see in practice what errors are thrown when we have database connection problems, but I fully agree that the most likely exception is indeed IOException

Reply:
indeed, let's keep an eye on that.